tests/message/trailer.c

```diff
@@ -0,0 +1,200 @@
+#include "clar_libgit2.h"
+#include "message.h"
+
+struct trailer {
+	const char *key;
+	const char *value;
+};
+
+struct cb_state {
+	struct trailer *trailer;
+};
+
+static int trailer_cb(const char *key, const char *value, void *st_)
+{
+	struct cb_state *st = st_;
+
+	cl_assert_equal_s(st->trailer->key, key);
+	cl_assert_equal_s(st->trailer->value, value);
+
+	st->trailer++;
+
+	return 0;
+}
+
+static void assert_trailers(const char *message, struct trailer *trailers)
+{
+	struct cb_state st = { trailers };
+
+	int rc = git_message_trailers(message, trailer_cb, &st);
+
+	cl_assert_equal_s(NULL, st.trailer->key);
+	cl_assert_equal_s(NULL, st.trailer->value);
+
+	cl_assert_equal_i(0, rc);
+}
+
+void test_message_trailer__simple(void)
+{
+	struct trailer trailers[3];
+
+	trailers[0].key   = "Signed-off-by";
+	trailers[0].value = "foo@bar.com";
```

Comment:
I think we can still use braced array initializers in ANSI C. Something like this should do the trick:

```c
struct trailer trailers[] = {
  { "Signed-off-by", "foo@bar.com" },
  /* ... */
  { NULL, NULL },
};
```

Reply:
Yep :+1: